pkg/validate/validate_test.go

```diff
@@ -327,3 +327,109 @@ func TestIsAzureZone(t *testing.T) {
 		t.Errorf("Valid FQDN failed to pass test: %s", validFqdn)
 	}
 }
+
+func TestValidateUpdateContainerService(t *testing.T) {
+	var twoAgents = []byte(`---
+location: eastus
+name: openshift
+properties:
+  routerProfiles:
+  - name: default
+    publicSubdomain: test.example.com
+  agentPoolProfiles:
+  - name: infra
+    role: infra
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: myCompute
+    role: compute
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+`)
+	var threeAgents = []byte(`---
```

Comment:
This is a bit of a confusing test because this manifest would not pass validate (we currently do not permit multiple compute agentpools).  I feel like the test might be better off calling Validate() but checking the errors returned against those expected from validateUpdateContainerService.  Or at least ensuring that the input manifests are valid in the first place.